presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/BaseCostBasedPlanTest.java

```diff
@@ -0,0 +1,245 @@
+/*
```

Comment:
I'm obviously not going through all the plan output files for tpc-h and tpc-ds so I'm just going to assume someone verified that the plans are good. 

I have some concerns about this testing style though.  It seems very brittle and frustrating to update for intentional changes.  I like the idea of having a quick way to know whether you possibly broke tpch or tpcds, but I think it will become more trouble than it's worth.  

I'm also not sure why this doesn't use the existing plan matching framework.  Is it because this is simpler and someone autogenerated the result files?

Reply:
I'm also a little bit afraid of s burden introduced by this PR. If something changes in the plan structure we would need to analyze all the failures, and fix over 9000 text files. 

Or, as it will happen more likely - we will be just blindly regenerating the test results (we end up doing this for plan matching tests in product tests at Teradata). That kinda kills the purpose of a test. 

I would really try to hard about whats the purpose of this test. And if the value worth the maintenance costs.

Reply:
Those tests are valuable as they allow us to catch real world regressions with supposedly innocent CBO/stats changes. Consider them as product tests that test CBO end-to-end. In general, any CBO changes should be benchmarked. We've seen changes that should improve stats calculations, but caused performance regressions.

Without framework here you would need to benchmark some set of queries to see if performance has changed. With this PR, you can focus on just a subset of queries and you can also validate if changed plans are rational. This greatly simplifies process of evaluating CBO modifications but also provides an automated way of catching potential regressions (we've been kicked with such case once).

One drawback though is that in order to validate performance changes one need to recreate real environment used for generating this plans. Currently it assumes 9 machines and we run actually benchmarks against them. So the question is how we provide standardized performance testbed so that we can validate CBO decisions against real query times.

Reply:
I don't have a strong opinion. Perhaps you can keep them for now. Let's see how often will we need to change them in practice.

Reply:
I would rather not have them here until it's easily actionable for people if they change.  If I don't have a way to evaluate whether a plan change caused a regression, I'm just going to blindly change the results.

Reply:
That's ok as long as you know what you're doing (you will know) and also, you know _that_ you are doing this (you will know because this of this test).

Reply:
but i wouldn't actually check for a regression because I wouldn't have the cluster set up to to do that, so it's just adding overhead.  It's one (or 100) more things to change without really helping with identifying problems.